manifests/server/rsync_shares.pp

```diff
@@ -49,78 +53,101 @@
+            path        => "${rsync_base}/${_env}/${_globals_dir}/mcafee",
+            hosts_allow => $_trusted_nets
+          }
+        }
 
-      rsync::server::section { "tftpboot_${_env}":
-        auth_users  => ["tftpboot_rsync_${_env}"],
-        comment     => "Tftpboot server configurations for Environment ${_env}",
-        path        => "${rsync_base}/${_env}/${_rsync_subdir}/tftpboot",
-        hosts_allow => $_trusted_nets
+        if 'jenkins_plugins' in $rsync_environments[$_env]['rsync']['global']['shares'] {
+          rsync::server::section { "jenkins_plugins_${_env}":
+            comment     => "Jenkins Configuration for Environment ${_env}",
+            path        => "${rsync_base}/${_env}/${_globals_dir}/jenkins_plugins",
+            hosts_allow => $_trusted_nets
+          }
+        }
       }
 
-      rsync::server::section { "mcafee_${_env}":
-        comment     => "McAfee DAT files for Environment ${_env}",
-        path        => "${rsync_base}/${_env}/mcafee",
-        hosts_allow => $_trusted_nets
-      }
+      # OS Specific Items
+      (keys($rsync_environments[$_env]['rsync']) - ['global','id']).each |String $_os| {
+        $_os_id = $rsync_environments[$_env]['rsync'][$_os]['id']
 
-      rsync::server::section { "clamav_${_env}":
-        comment     => "ClamAV Virus Database Updates for Environment ${_env}",
-        path        => "${rsync_base}/${_env}/clamav",
-        hosts_allow => $_trusted_nets
-      }
+        # OS Major Version Specific Items
+        (keys($rsync_environments[$_env]['rsync'][$_os]) - ['global','id']).each |String $_os_maj_ver| {
+          $_os_maj_ver_id = $rsync_environments[$_env]['rsync'][$_os][$_os_maj_ver]['id']
 
-      rsync::server::section { "dhcpd_${_env}":
-        auth_users  => ["dhcpd_rsync_${_env}"],
-        comment     => "DHCP Configurations for Environment ${_env}",
-        path        => "${rsync_base}/${_env}/${_rsync_subdir}/dhcpd",
-        hosts_allow => $_trusted_nets
-      }
+          if 'bind_dns' in $rsync_environments[$_env]['rsync'][$_os][$_os_maj_ver]['shares'] {
+            rsync::server::section { "bind_dns_default_${_env}_${_os_id}_${_os_maj_ver_id}":
+              auth_users  => ["bind_dns_default_rsync_${_env}_${_os_id}_${_os_maj_ver_id}"],
+              comment     => "Default DNS configurations for named for Environment ${_env} on ${_os_id} ${_os_maj_ver_id}",
+              path        => "${rsync_base}/${_env}/rsync/${_os_id}/${_os_maj_ver_id}/bind_dns/default",
+              hosts_allow => $_trusted_nets
+            }
+          }
+        }
 
-      rsync::server::section { "snmp_${_env}":
-        comment     => "SNMP MIBs and Modules for Environment ${_env}",
-        path        => "${rsync_base}/${_env}/${_rsync_subdir}/snmp",
-        hosts_allow => $_trusted_nets
-      }
+        # OS Global Items
+        if 'apache' in $rsync_environments[$_env]['rsync'][$_os]['global']['shares'] {
```

Comment:
It was failing on this until I changed 'global' to 'Global', saying something about it being empty.